product-web-app/service/jobRoleSingleViewService.ts

```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+import JobRole from '../model/jobRoleSingleView.js';
+import logger from './logger.js';
+
+
+export default class JobRoleService {
+
+  async getJobRoleSpecification(id: number): Promise<JobRole> {
+    try {
+      const response = await axios.get('http://localhost:8080/api/jobRoles/' + id,);
```

Comment:
Localhost is redudent, we set base url in app.ts
```
axios.defaults.baseURL = API_URL;
```

```suggestion
      const response = await axios.get(API.GET_JOB_ROLES(id));
```